### PR TITLE
Full compile of some minimizers

### PR DESCRIPTION
**Pull request by StephenNneji**
All minimizer are included except for the nested sampler

### CODE REVIEW

File: minimisers/generalUtils/bayesStats/refPrctileConfInts.m [hunk 1]
```diff
@@ -118,29 +148,48 @@
...
+end
+
+function z = makeCell(n,vals)
+%#codegen
+% assert(n < 100);
+x = cell(n,1);   
+for i = 1:n
+    x{i} = vals;
+end
+z = x;
```
Comment (DrPaulSharp) [inline minimisers/generalUtils/bayesStats/refPrctileConfInts.m:183]: Why not just use z throughout this routine?

File: minimisers/generalUtils/bayesStats/refPrctileConfInts.m [hunk 2]
```diff
@@ -118,29 +148,48 @@
...
+
+function z = makeCell2D(n,vals)
+%#codegen
+% assert(n < 100);
+x = cell(n,2);   
+for i = 1:n
+    x{i,1} = vals;
+    x{i,2} = vals;
+end
+z = x;
```
Comment (DrPaulSharp) [inline minimisers/generalUtils/bayesStats/refPrctileConfInts.m:194]: . . . and this one?

File: examples/D2O/testScript.m [hunk 3]
```diff
@@ -0,0 +1,48 @@
+% THIS FILE IS GENERATED FROM RAT VIA THE "WRITESCRIPT" ROUTINE. IT IS NOT PART OF THE RAT CODE.
```
[the remark this answers is not in the document]
Comment (StephenNneji) [reply examples/D2O/testScript.m:1]: No I missed it

File: API/textProgressBar.m [hunk 4]
```diff
@@ -18,12 +20,21 @@ function textProgressBar(curm,pct)
...
+fprintf('%s%s',repmat(sprintf('\b'),1,lastNchar),progressmsg);
 drawnow
 lastNchar=length(progressmsg);
 
 end
 
+%     c = 50;
+%     strDotsMax = 20;
+%     nDots = floor(c/*strDotsMax);
+%     dotOut = ['[' repmat('*',1,nDots) repmat('.',1,strDotsMax-nDots) ']']
```
[the remark this answers is not in the document]
Comment (StephenNneji) [reply API/textProgressBar.m:40]: I have no idea

File: minimisers/DREAM/functions/makeEmptyBayesResultsStruct.m [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+
+% Make the final structure...
+bayesResults = struct('bestFitsMean',bestFitsMean,...
+                      'predlims', predlims,...
+                      'parConfInts', parConfInts,...
+                      'bestPars', bestPars,...
+                      'bayesRes', bayesRes,...
+                      'chain',chain);
+
+end
```
[the remark this answers is not in the document]
Comment (StephenNneji) [reply minimisers/DREAM/functions/makeEmptyBayesResultsStruct.m:274]: I will refactor but there are a ton of comments and commented code which I hope to clean-up once we release the beta

File: API/parseClassToStructs.m [hunk 6]
```diff
@@ -210,10 +210,8 @@
             priorType = 1;
         elseif strcmpi(thisType,'gaussian')
             priorType = 2;
-        elseif strcmpi(thisType,'jeffreys')
-            priorType = 3;
         else
-            priorType = 1;
+            priorType = 3;
```
Comment (DrPaulSharp) [inline API/parseClassToStructs.m:214]: The original code implied that there may be a case other than "uniform", "gaussian", "jeffreys". I take it this isn't the case?

**Review by DrPaulSharp — APPROVED** [review]

**Pull request merged**